src/ott/neural/flow_models/flows.py

```diff
@@ -0,0 +1,164 @@
+      src: Sample from the source distribution of shape ``[batch, ...]``.
+      tgt: Sample from the target distribution of shape ``[batch, ...]``.
+    """
+
+  @abc.abstractmethod
+  def compute_sigma_t(self, t: jnp.ndarray) -> jnp.ndarray:
+    """Compute the standard deviation of the probability path at time :math:`t`.
+
+    Args:
+      t: Time :math:`t` of shape ``[batch, 1]``.
+
+    Returns:
+      Standard deviation of the probability path at time :math:`t`.
+    """
+
+  @abc.abstractmethod
+  def compute_ut(
+      self, t: jnp.ndarray, src: jnp.ndarray, tgt: jnp.ndarray
+  ) -> jnp.ndarray:
+    """Evaluate the conditional vector field.
+
+    Evaluate the conditional vector field defined between :math:`x_0` and
+    :math:`x_1` at time :math:`t`.
+
+    Args:
+      t: Time :math:`t` of shape ``[batch, 1]``.
+      src: Sample from the source distribution of shape ``[batch, ...]``.
+      tgt: Sample from the target distribution of shape ``[batch, ...]``.
+
+    Returns:
+      Conditional vector field evaluated at time :math:`t`.
+    """
+
+  def compute_xt(
+      self, rng: jax.Array, t: jnp.ndarray, src: jnp.ndarray, tgt: jnp.ndarray
+  ) -> jnp.ndarray:
+    """Sample from the probability path.
+
+    Sample from the probability path between :math:`x_0` and :math:`x_1` at
+    time :math:`t`.
+
+    Args:
+      rng: Random number generator.
+      t: Time :math:`t` of shape ``[batch, 1]``.
+      src: Sample from the source distribution of shape ``[batch, ...]``.
+      tgt: Sample from the target distribution of shape ``[batch, ...]``.
+
+    Returns:
+      Samples from the probability path between :math:`x_0` and :math:`x_1`
+      at time :math:`t`.
+    """
+    noise = jax.random.normal(rng, shape=src.shape)
+    mu_t = self.compute_mu_t(t, src, tgt)
+    sigma_t = self.compute_sigma_t(t)
+    return mu_t + sigma_t * noise
+
+
+class StraightFlow(BaseFlow, abc.ABC):
+  """Base class for flows with straight paths.
+
+  Args:
+    sigma: Noise used for computing time-dependent noise schedule.
+  """
+
+  def compute_mu_t(  # noqa: D102
+      self, t: jnp.ndarray, src: jnp.ndarray, tgt: jnp.ndarray
+  ) -> jnp.ndarray:
+    return (1.0 - t) * src + t * tgt
+
+  def compute_ut(  # noqa: D102
+      self, t: jnp.ndarray, src: jnp.ndarray, tgt: jnp.ndarray
+  ) -> jnp.ndarray:
+    del t
+    return tgt - src
+
+
+class ConstantNoiseFlow(StraightFlow):
+  r"""Flow with straight paths and constant flow noise :math:`\sigma`.
+
+  Args:
+    sigma: Constant noise used for computing time-independent noise schedule.
+  """
+
+  def compute_sigma_t(self, t: jnp.ndarray) -> jnp.ndarray:
+    r"""Compute noise of the flow at time :math:`t`.
+
+    Args:
+      t: Time :math:`t` of shape ``[batch, 1]``.
+
+    Returns:
+      Constant, time-independent standard deviation :math:`\sigma`.
+    """
+    return jnp.full_like(t, fill_value=self.sigma)
+
+
+class BrownianNoiseFlow(StraightFlow):
```

Comment:
This is most commonly known as the Brownian bridge, https://en.wikipedia.org/wiki/Brownian_bridge